tests/unit/test-virtio-dmabuf.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * QEMU tests for shared dma-buf API
+ *
+ * Copyright (c) 2023 Red Hat, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "qemu/osdep.h"
+#include "hw/virtio/virtio-dmabuf.h"
+
+
+static void test_add_remove_resources(void)
+{
+    QemuUUID uuid;
+    int i, dmabuf_fd;
+
+    for (i = 0; i < 100; ++i) {
+        qemu_uuid_generate(&uuid);
+        dmabuf_fd = g_random_int_range(3, 500);
+        /* Add a new resource */
+        g_assert(virtio_add_dmabuf(uuid, dmabuf_fd));
+        g_assert_cmpint(virtio_lookup_dmabuf(uuid), ==, dmabuf_fd);
+        /* Remove the resource */
+        g_assert(virtio_remove_resource(uuid));
+        /* Resource is not found anymore */
+        g_assert_cmpint(virtio_lookup_dmabuf(uuid), ==, -1);
+    }
+}
+
+
+static void test_remove_invalid_resource(void)
+{
+    QemuUUID uuid;
+    int i;
+
+    for (i = 0; i < 20; ++i) {
+        qemu_uuid_generate(&uuid);
+        g_assert(virtio_lookup_dmabuf(uuid) == -1);
```

Comment:
Not familiar with how this testing framework works.
Is every test function starts from a new environment?
If so, in this test flow you always try to lookup at / remove from an empty lookup table and might be worth (**additionally** to this scenario) to fill it with UUIDs A, B, C and try to lookup / remove UUIDs X, Y, Z.

Reply:
`Is every test function starts from a new environment?` yes

`to fill it with UUIDs A, B, C and try to lookup / remove UUIDs X, Y, Z.` I'll try to add that. Also noticed that this should be `g_assert_cmpint` instead :)